cmd/clusterctl/client/cluster/objectgraph.go

```diff
@@ -176,12 +186,40 @@ type objectGraph struct {
 
 func newObjectGraph(proxy Proxy, providerInventory InventoryClient) *objectGraph {
 	return &objectGraph{
+		externalObjects:   []*node{},
 		proxy:             proxy,
 		providerInventory: providerInventory,
 		uidToNode:         map[types.UID]*node{},
 	}
 }
 
+// addExternal adds an external Kubernetes object to the object graph.
+func (o *objectGraph) addExternal(obj *unstructured.Unstructured) error {
+	if _, found := o.uidToNode[obj.GetUID()]; found {
+		return nil
+	}
+
+	kind, apiVersion := obj.GetObjectKind().GroupVersionKind().ToAPIVersionAndKind()
+	node := &node{
+		identity: corev1.ObjectReference{
+			Name:       obj.GetName(),
+			Namespace:  obj.GetNamespace(),
+			UID:        obj.GetUID(),
+			Kind:       kind,
+			APIVersion: apiVersion,
+		},
+		additionalInfo: make(map[string]interface{}),
```

Comment:
Am I wrong or without calling captureAdditionalInformation we won't get values for clusterTopologyNameKey & clusterTopologyNamespaceKey.